src/app/common/search-select/search-select.component.ts

```diff
@@ -9,28 +9,60 @@ import { SkillService } from 'src/app/skill.service';
   styleUrls: ['./search-select.component.scss']
 })
 export class SearchSelectComponent implements OnInit {
-  constructor(private skillService: SkillService) {}
+  constructor(private skillService: SkillService,
+    private eRef: ElementRef) {
+    this.allSkills = [];
+    this.skillExistsEmitter.emit(this.skillExists);
+  }
 
   selections = new FormControl();
   @Output() emitSelections = new EventEmitter<any>();
   title = 'Skills';
   options: Skill[];
   allSkills: Skill[];
+  @Output() skillExistsEmitter = new EventEmitter<any>();
+  skillExists = true;
+  @Output() skillTitleEmitter = new EventEmitter<any>();
+  searchTerm: string = "";
+
 
   selectOption() {
     this.emitSelections.emit(this.selections);
   }
 
+  // @HostListener('document:click', ['$event'])
```

Comment:
Why is this left as a comment? If it is not needed, it should be removed. Additionally, this appears to be the only use of HostListener. Since the only time it is used is commented out, then the import should also be removed.